File: app/Models/ProductoTerminado.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ProductoTerminado extends Model
{
    use HasFactory;

    protected $table = 'producto_terminado';
    protected $primaryKey = 'id_producto_terminado';

    protected $fillable = [
        'nombre',
        'descripcion',
        'categoria',
        'costo',
        'precio',
        'stock_actual',
        'stock_min',
        'estado',
        'fecha_ingreso',
        'id_producto'
    ];

    protected $casts = [
        'fecha_ingreso' => 'date',
        'costo' => 'decimal:2',
        'precio' => 'decimal:2',
    ];

    public function producto()
    {
        return $this->belongsTo(Producto::class, 'id_producto', 'id_producto');
    }
}

File: src/main/java/com/Wurger/model/ProductoTerminado.java
package com.Wurger.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "producto_terminado")
@Getter
@Setter
public class ProductoTerminado {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_producto_terminado")
    private Integer id;

    @Column(nullable = false, length = 100)
    private String nombre;

    private String descripcion;

    @Column(length = 50)
    private String categoria;

    private BigDecimal costo;

    private BigDecimal precio;

    @Column(name = "stock_actual")
    private Integer stockActual;

    @Column(name = "stock_min")
    private Integer stockMin;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private Estado estado;

    @Column(name = "fecha_ingreso")
    private LocalDate fechaIngreso;

    @Column(length = 255)
    private String imagen;

    // Relación con Recetas
    // Aquí NO usamos JsonIgnore porque es útil ver los ingredientes del plato.
    @OneToMany(mappedBy = "productoTerminado", cascade = CascadeType.ALL)
    private List<Receta> recetas;

    public enum Estado {
        Activo,
        Inactivo
    }
}